tanjun/checks.py

```diff
@@ -787,3 +788,63 @@ def with_check(check: tanjun_abc.CheckSig, /) -> collections.Callable[[CommandT]
         A command decorator callback which adds the check.
     """
     return lambda command: command.add_check(check)
+
+
+def with_any_role_check(
+    roles: list[str | int] = list(),
+    *,
+    error_message: str | None = "You do not have the required roles to use this command!",
+    halt_execution: bool = True,
+) -> collections.Callable[[CommandT], CommandT]:
+    """Only let a command run if the author has a specific role.
+
+    Parameters
+    ----------
+    roles: list[str | int]
+        The author must have at least one (1) role in this list. (Role.name and Role.id are checked)
+
+    Other Parameters
+    ----------------
+    error_message: str | None
+        The error message raised if the member does not have a required role.
+
+        Defaults to 'You do not have the required roles to use this command!'
+    halt_execution: bool
+        Whether this check should raise `tanjun.errors.HaltExecution` to
+        end the execution search when it fails instead of returning `False`.
+
+        Defaults to `False`.
+
+    Returns
+    -------
+    collections.abc.Callable[[CommandT], CommandT]
+        A command decorator callback which adds the check."""
+
+    def any_role_check(
+        ctx: tanjun_abc.Context,
+        *,
+        roles: list[str],
+        error_message: str | None = None,
+        halt_execution: bool = False,
+    ) -> bool:
+
+        if not ctx.member:
```

Comment:
```suggestion
        if not ctx.is_human:
```

Reply:
I was actually checking `ctx.member` to also ensure that the command was not executed in a DM, as then `member` would be set to `None`. Would `is_human` be preferable?

Reply:
So you'll also need to check if the `guild_id` is also not `None` in this case.

Reply:
generally speaking these checks shouldn't leak into each other, if they also want to enforce that the author should be human then they can add that as a separate check.

Reply:
I was doing this check to ensure that `ctx.member` existed so that I would still be able to access the members role. If a non-member called this, (or from a dm) it would raise an attribute error for `ctx.member` not being set, if I am reading the docs correctly.

I was primarily using AuthorPermissionsCheck as a template. I see that `get_permission` does a similar check, but then provides a default Permission set instead. This Check would be better to just err outright and halt execution, imo.

Reply:
yeah checking for ctx.member is fine, just checking if they're human is out of scope